proxy/src/main/java/com/velocitypowered/proxy/plugin/util/PluginDependencyUtils.java

```diff
@@ -0,0 +1,67 @@
+package com.velocitypowered.proxy.plugin.util;
+
+import com.google.common.graph.Graph;
+import com.google.common.graph.GraphBuilder;
+import com.google.common.graph.MutableGraph;
+import com.velocitypowered.api.plugin.PluginCandidate;
+import com.velocitypowered.api.plugin.meta.PluginDependency;
+
+import java.util.*;
+
+public class PluginDependencyUtils {
+    public static List<PluginCandidate> sortCandidates(List<PluginCandidate> candidates) {
+        // Create our graph, we're going to be using this for Kahn's algorithm.
+        MutableGraph<PluginCandidate> graph = GraphBuilder.directed().allowsSelfLoops(false).build();
+
+        // Add edges
+        for (PluginCandidate description : candidates) {
+            graph.addNode(description);
+
+            for (PluginDependency dependency : description.getDependencies()) {
+                Optional<PluginCandidate> in = candidates.stream().filter(d -> d.getId().equals(dependency.getId())).findFirst();
```

Comment:
If a plugin depends on a lot of dependencies and the user has lots of plugins installed, this could slow down quickly. Consider having an ID->candidate map?